src/socket_tls_impl.cpp

```diff
@@ -91,22 +177,30 @@ void SocketTlsImpl::SslDeleter::operator()(SSL *ptr) const noexcept
 SocketTlsImpl::SocketTlsImpl(int family, int type, int protocol,
     char const *certFilePath, char const *keyFilePath)
   : SocketImpl(family, type, protocol)
-  , sslGuard()
+  , sslGuard() // must be created before call to SSL_CTX_new
```

Comment:
All this repetition is annoying; a delegating constructor would be nicer. But since the guard has to be initialized beforehand that is not so easy.